msteams-platform/toolkit/add-resource.md

```diff
@@ -8,81 +8,91 @@ ms.topic: overview
 ms.date: 11/29/2021
 ---
 
-# Add cloud resources to your Teams app
+# Add cloud resources to Teams app
 
-TeamsFx helps to provision cloud resources for your application hosting. You can also optionally add cloud resources that fit your development needs.
+TeamsFx helps to provision the cloud resources for your application hosting. You can add the cloud resources optionally, that fit your development needs.
 
-## Prerequisite
+## Advantages
 
-[Install Teams Toolkit](https://marketplace.visualstudio.com/items?itemName=TeamsDevApp.ms-teams-vscode-extension) version v3.0.0+.
+The following list provides advantages to add more cloud resources in TeamsFx:
 
-> [!TIP]
-> Ensure you have Teams app project in Visual Studio Code.
+* Provides convenience
+* Autogenerates all configuration files and connect to Teams app by using Teams Toolkit
 
-## Add cloud resources using Teams Toolkit
+## Limitation
+
+If you have created SPFx based tab project, you can't add the resources.
```

Comment:
should the limitation section to be put in the end of this doc?
and the if user have created SPFx based tab project, they cannot add "Azure cloud resources".  "can't add resources " will feraten the user.

Reply:
It is not required to change limitation section because we followed same flow in all the similar documents. Remaining changes updated as mentioned.

Reply:
Hi @zyxiaoyuer, could we discuss on this on Monday.